### DISCUSSION TITLE
Fails silently when `delete`ing a non-empty folder on Windows

### DISCUSSION THREAD

**Discussion by bicarlsen** [Byron/trash-rs#69] [Q&A]
On Windows, it seems the `delete` function does not delete a directory if it has contents, but will return `Ok`. Is this intended behavior? If so, would it be worth adding a `delete_recursive` convenience function so users don't have to manually walk the directory tree?

**Comment by Byron** [comment]
Thanks for bringing this up. 

When looking at the documentation of [delete()](https://docs.rs/trash/3.0.2/trash/struct.TrashContext.html#method.delete) it's not entirely clear what should happen when a directory is encountered. It's likely that 'single' means 'empty' in case of a directory. This could be validated by looking at the tests, and then an improvement should be made to the docs to clarify.

Based on that outcome, a `delete_recursive()` function might be needed to handle this case more conveniently.
